### PR TITLE
Cw1155 Approve/Revoke single

### PR DESCRIPTION
**Pull request by dirtyshab**
This implements a `token_approves` Map in order to store approvals on specific tokens/amounts that a user owns. It is expected to work the same as approvals on cw721 but with an amount argument to limit the quantity of the token the operator has access to.

### CODE REVIEW

File: contracts/cw1155-base/src/lib.rs [hunk 1]
```diff
@@ -9,6 +9,9 @@ pub use crate::state::Cw1155Contract;
 use cosmwasm_std::Empty;
 use cw1155::Cw1155QueryMsg;
 
+// todo - is this standard?
+pub const CW_ADDRESS_LENGTH: usize = 62;
```
Comment (dirtyshab) [inline contracts/cw1155-base/src/lib.rs:13]: todo - check this: used for deserializing storage key

File: packages/cw1155/src/event.rs [hunk 2]
```diff
@@ -21,13 +20,19 @@ impl TransferEvent {
...
 impl From<TransferEvent> for Event {
     fn from(event: TransferEvent) -> Self {
-        Event::new("transfer_tokens").add_attributes(
-            vec![
-                attr("sender", event.sender.as_str()),
-                attr("recipient", event.recipient.as_str()),
-                attr("tokens", event.tokens.iter().map(|t| t.to_string()).collect::<Vec<_>>().join(",")),
-            ]
-        )
+        Event::new("transfer_tokens").add_attributes(vec![
```
Comment (dvli2007) [inline packages/cw1155/src/event.rs:23]: @apolloshab should we name this `transfer_nfts` to be more consistent with the current CW721 standard?

Comment (dirtyshab) [reply packages/cw1155/src/event.rs:23]: agreed, thinking of changing to *_single and *_batch for everything

File: contracts/cw1155-base/src/lib.rs (hunk 1 above)
Comment (dirtyshab) [reply contracts/cw1155-base/src/lib.rs:13]: nvm, removing TokenKey since it's unnecessary - using token id as primary key instead

**Pull request merged**